flag/flags.go

```diff
@@ -0,0 +1,12 @@
+package flag
+
+// Flags provides the ability to create and communicate feature flags.
+type Flags uint64
+
+// Flags is a bitset
+const (
+	Debug Flags = 1 << iota
+	SamplingSet
```

Comment:
ok. makes sense! Might be worth copying some docs just so folks know.

For example, in all carriers (even http), debug is read. In others they are
not (which can be documented by stealing some notes from here)
https://github.com/openzipkin/brave/blob/master/brave-core/src/main/java/com/github/kristofa/brave/SpanId.java

Note: no zipkin http carrier I know of expects a value > 1 for flags at the
moment. So, if something >1 is serialized into the http header, it might
not work with instrumentation that try to read 0 or 1.
This might lead to unexpected behavior (even if such a value shouldn't
fail). I'd advise only serializing 0 or 1 for flags, in the case of http,
for now. I can help work through other language instrumentation so that
eventually values besides 0 or 1 work.